evm_arithmetization/src/cpu/kernel/asm/memory/syscalls.asm

```diff
@@ -225,6 +225,12 @@ global sys_mcopy:
     // stack: kexit_info, dest_offset, offset, size
     %wcopy_charge_gas
 
+    // stack: kexit_info, dest_offset, offset, size
+    DUP2
+    %ensure_offset_in_range
```

Comment:
I think this is somewhat redundant with the call to `%ensure_reasonable_offset` below.

Reply:
Hmm yeah `%ensure_reasonable_offset` would be triggered even before we get an `offset` overflow